agent/bench.go

```diff
@@ -548,9 +561,31 @@ func (b *Bench) RerunKube(cmd, cmdRemap string, forced bool) {
 		b.kubeCisCmds[cmd] = cmdRemap
 	}
 
+	// Check if the node is a Kubernetes master
 	_, b.isKubeMaster = b.kubeCisCmds[cmdKubeApiServer]
+
+	// Check if the node is a Kubernetes worker
 	_, b.isKubeWorker = b.kubeCisCmds[cmdKubelet]
 
+	k8sVer, _ := global.ORCH.GetVersion(false, false)
+
+	// Only fall back to K3s detection if the node is neither a master nor a worker
+	if !b.isKubeMaster && !b.isKubeWorker && strings.Contains(k8sVer, "k3s") {
+		// use dir := "/proc/1/root/var/lib/rancher/k3s/server" to check if in k3s master
+		dir := "/proc/1/root/var/lib/rancher/k3s/server"
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
+			b.isKubeWorker = true
+			b.kubeCisCmds[cmdKubelet] = cmdKubelet
+			b.isK3s = true
+		} else if err == nil {
+			b.isKubeMaster = true
+			b.kubeCisCmds[cmdKubeApiServer] = cmdKubeApiServer
+			b.isK3s = true
+		} else {
+			log.WithFields(log.Fields{"error": err, "directory": dir}).Error("Error checking directory")
+		}
+	}
```

Comment:
I like the way you get k8s version to make sure it's a k3s.
1. With false and false in GetVersion's argument, do you get a valid `k8sVer`?
2. minor: As a better practice, you probably should leave `b.isKubeWorker` and `b.isKubeMaster` intact (line 564-568) until you make the final decision about its platform?
3. Can k3s have both server and worker on the same node?

Reply:
Hi @holyspectral ,
Thanks for review.

1. With false and false in GetVersion's argument, do you get a valid k8sVer? yes
2. minor: As a better practice about this, I think line 573-586 shows it's k3s, thus this should be fine?
3. if this is a single node cluster, then it's both server and worker node.

Reply:
about 3, if this, I think we should have another case for this?

Reply:
2 is more about coding style.  When you update `b.isKubeMaster` and `b.isKubeWorker` before the logic completes, it's prone to leave the two fields in an unexpected state if the error handling is not implemented correctly.  One example is when you have to exit the function with an unrecoverable error. 

A better way is like this:
```
	// Keep the variables in local variables for now,
	_, isKubeMaster := b.kubeCisCmds[cmdKubeApiServer]
	_, isKubeWorker := b.kubeCisCmds[cmdKubelet]
	k8sVer, _ := global.ORCH.GetVersion(false, false)

	// Only fall back to K3s detection if the node is neither a master nor a worker
	if !isKubeMaster && !isKubeWorker && strings.Contains(k8sVer, "k3s") {
            // determine the value for k3s.
	}
	
        // assign the value
	b.isKubeMaster = isKubeMaster
	b.isKubeWorker = isKubeWorker
	b.isK3s = isK3s
```

So even if the function has to exit early, the state in this struct won't be changed.  

For 3, agree.  We can get this merged first.

Reply:
Hi @holyspectral ,
Just found the github question about if the k3s node can be server and agent at the same time. https://github.com/k3s-io/k3s/issues/1279
I think we can resolve it in this PR as well

What do you think?

Reply:
Yeah it sounds good to me.

Reply:
please take a look, thanks